python/lsst/dax/imgserv/jobqueue/imageworker.py

```diff
@@ -78,17 +80,30 @@ def get_image_async(self, *args, **kwargs):
     config["DAX_IMG_CONFIG"] = config_path
     meta_url = imgserv_config.webserv_config["dax.imgserv.meta.url"]
     config["DAX_IMG_META_URL"] = meta_url
-    soda = imageSODA.ImageSODA(config)
-    image = soda.do_sync(params)
-    # save the result image in local temp dir
-    with tempfile.NamedTemporaryFile(dir=config["DAX_IMG_TEMPDIR"],
+    image = None; status = "OK"
+    try:
+        soda = imageSODA.ImageSODA(config)
+        image = soda.do_sync(params)
+    except ImageNotFound:
+        status = "Error=Image Not Found"
+    except UsageError as e:
+        status = "Error="+e.message
+    except Exception as e:
+        status = str(e)
+    finally:
+        # save the result image in local temp dir
+        if image:
+            with tempfile.NamedTemporaryFile(dir=config["DAX_IMG_TEMPDIR"],
```

Comment:
It seems like even though you're in the finally, you're not actually done.  A lot can still go wrong, for example, out of disk space.  It seems like all of this part should be in the try.  (All the way up until job_end_time).

Reply:
Very good point, taken.